Assets/__Scripts/Map/BeatmapObject.cs

```diff
@@ -100,4 +102,5 @@ public virtual bool IsConflictingWith(BeatmapObject other, bool deletion = false
         }
         return false;
     }*/
+    public abstract void Apply(BeatmapObject originalData);
```

Comment:
I think some code could be re-used here if you made this method `virtual`. We could have the `_time` and `_customData` applications be done here, and the overriding methods could just call `base.Apply(originalData);` to re-use this code. Just personal preference, though

Reply:
Done